basic_hashtable/b_hashtables.c

```diff
@@ -114,6 +132,14 @@ char *hash_table_retrieve(BasicHashTable *ht, char *key)
  ****/
 void destroy_hash_table(BasicHashTable *ht)
 {
+  for(int i = 0; i < ht->capacity; i++)
+  {
+    if (!ht->storage[i] == NULL)
```

Comment:
this is awkward and causes warnings from the compiler
I think `if (ht->storage[i] != NULL)` accomplishes what you want